app/src/main/java/org/torproject/android/ui/onboarding/CustomSlideBigText.java

```diff
@@ -57,24 +62,48 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
     public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
                              @Nullable Bundle savedInstanceState) {
         View view = inflater.inflate(layoutResId, container, false);
-        ((TextView)view.findViewById(R.id.custom_slide_big_text)).setText(mTitle);
-
+        title = ((TextView) view.findViewById(R.id.custom_slide_big_text));
+        title.setText(mTitle);
+        tv = (TextView) view.findViewById(R.id.custom_slide_big_text_sub);
         if (!TextUtils.isEmpty(mSubTitle)) {
 
-            TextView tv =
-                    (TextView)view.findViewById(R.id.custom_slide_big_text_sub);
             tv.setText(mSubTitle);
             tv.setVisibility(View.VISIBLE);
         }
 
-        if (mButtonText != null)
-        {
-            Button button = (Button)view.findViewById(R.id.custom_slide_button);
+        if (mButtonText != null) {
+            button = (Button) view.findViewById(R.id.custom_slide_button);
             button.setVisibility(View.VISIBLE);
             button.setText(mButtonText);
             button.setOnClickListener(mButtonListener);
         }
         return view;
 
     }
+
+    //Restoring the data
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        if (savedInstanceState != null) {
+            title.setText(savedInstanceState.getString(getResources().getString(R.string.Title)));
+            tv.setText(savedInstanceState.getString(getResources().getString(R.string.SubTitle)));
+            if (mButtonText != null) {
+                button.setText(savedInstanceState.getString(getResources().getString(R.string.ButtonText)));
+            }
+
+        }
+    }
+
+    //Saving the data
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putString(getResources().getString(R.string.Title), mTitle);
```

Comment:
I would change the names of these resource keys to pref_title or something so people don't change the value of the key and create duplicate preferences

Reply:
Okay, will do.